app/lib/frontend/backend.dart

```diff
@@ -1052,17 +1052,15 @@ class TarballStorageNamer {
             (namespace == null || namespace.isEmpty) ? '' : 'ns/$namespace/';
 
   /// The GCS object name of a tarball object - excluding leading '/'.
-  String tarballObjectName(String package, String version)
-      // TODO: Do we need some kind of escaping here?
-      =>
+  String tarballObjectName(String package, String version) =>
       '${prefix}packages/$package-$version.tar.gz';
 
   /// The GCS object name of an temporary object [guid] - excluding leading '/'.
   String tmpObjectName(String guid) => 'tmp/$guid';
 
   /// The http URL of a publicly accessable GCS object.
   String tarballObjectUrl(String package, String version) {
-    final object = tarballObjectName(package, version);
+    final object = tarballObjectName(package, Uri.encodeComponent(version));
```

Comment:
Are you sure this is a problem? why have anything been working before?

Have you tested this on staging...


Sorry for the fuzz, it just seems like something that we could easily double encode by accident.. and something that could totally break `pub get`.

Reply:
I'll test it in staging. Integration test does test the escaped URL and it works.

Reply:
I did run the integration test on staging, it looks good.